src/main/java/application/FormController.java

```diff
@@ -94,6 +95,20 @@ public NewFormResponse newFormResponse(@RequestBody NewSessionRequestBean newSes
         return newFormResponseFactory.getResponse(newSessionBean, postUrl);
     }
 
+    @ApiOperation(value = "Change the form session locale")
+    @RequestMapping(value = Constants.URL_CHANGE_LANGUAGE, method = RequestMethod.POST)
+    @UserLock
+    @UserRestore
+    @ConfigureStorageFromSession
+    public FormEntryResponseBean changeLocale(@RequestBody ChangeLocaleRequestBean changeLocaleBean,
+                                                @CookieValue(Constants.POSTGRES_DJANGO_SESSION_ID) String authToken) throws Exception {
+        SerializableFormSession serializableFormSession = formSessionRepo.findOneWrapped(changeLocaleBean.getSessionId());
```

Comment:
It looks like you don't really worry about this in other similar routes, but should we not protect against the possibility of this endpoint getting hit with an invalid or no-longer-present session ID, and therefore having no form session returned here?

Reply:
@amstone326 We handle that explicitly [here](https://github.com/dimagi/formplayer/blob/1258b69e83b887638881e815fbdc3ecd803faaff/src/main/java/repo/impl/PostgresFormSessionRepo.java#L192-L192) and reset the session in the front end

Reply:
Ah sorry, I should have looked a bit more at that. Thanks.